Modules/_ssl.c

```diff
@@ -2073,6 +2077,111 @@ cipher_to_dict(const SSL_CIPHER *cipher)
        );
 }
 
+/*[clinic input]
+_ssl._SSLSocket.get_ech_status
+[clinic start generated code]*/
+
+static PyObject *
+_ssl__SSLSocket_get_ech_status_impl(PySSLSocket *self)
+/*[clinic end generated code: output=263bf7fc7888e2d6 input=4f68a05d59f39372]*/
+{
+#ifdef OPENSSL_ECH
+    char *inner_sni = NULL, *outer_sni = NULL;
+
+    if (self->ssl == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    int status = SSL_ech_get1_status(self->ssl, &inner_sni, &outer_sni);
+	PyObject *retval = PyTuple_New(3);
+    PyTuple_SetItem(retval, 0, PyLong_FromLong(status));
+
+    if (inner_sni != NULL) {
+        PyTuple_SetItem(retval, 1, PyUnicode_FromString(inner_sni));
+        OPENSSL_free(inner_sni);
+    } else {
+        PyTuple_SetItem(retval, 1, Py_None);
+        Py_INCREF(Py_None);
+    }
+
+    if (outer_sni != NULL) {
+        PyTuple_SetItem(retval, 2, PyUnicode_FromString(outer_sni));
+        OPENSSL_free(outer_sni);
+    } else {
+        PyTuple_SetItem(retval, 2, Py_None);
+        Py_INCREF(Py_None);
+    }
+
+    return retval;
+#else
+    PyErr_SetString(PyExc_NotImplementedError, "OpenSSL does not have ECH support");
```

Comment:
I don't remember whether we raise a NotImplementedError or just plainly don't define the function at the module level.